extensions/opencv/CaptureRTSPFrame.cpp

```diff
@@ -73,6 +73,11 @@ void CaptureRTSPFrame::initialize() {
   relationships.insert(Success);
   relationships.insert(Failure);
   setSupportedRelationships(std::move(relationships));
+  // By default in OpenCV, ffmpeg capture is hardcoded to use TCP and this is a workaround
+  // also if UDP timeout, ffmpeg will retry with TCP
+  // Note:
+  // 1. OpenCV community are trying to find a better approach than setenv.
+  setenv("OPENCV_FFMPEG_CAPTURE_OPTIONS", "rtsp_transport;udp", 1);
```

Comment:
setenv is unfortunately not thread-safe. Not only setting the same environment variable from two threads is unsafe, but so is setting anything from two threads at once.
That being said, I know no good solution to this other than a framework for the global synchronization of setenvs and getenvs, and even that does not account from potential setenvs/getenvs called by third parties.
I don't think that there is a high probability of this causing issues, but if we see unexplainable crashes or issues around environment variables, then we should think of this as a cause.
I would like this comment to be extended with this caveat.

Reply:
Since this is an env option why not leave it in the operating system environment ( esp since this isn't cross platform compatible ). Can add something to minifi.sh . I have a feeling at some point we will have OPTS set via the conf dir that minifi.sh pulls and adds to a running instance. 

@bakaid this may be a case for using the Initializer construct you added if the shell script doesn't make sense

Reply:
@phrocker Yes, we have ended up with adding an ObjectFactoryInitializer for the OpenCV component with @nghiaxlee, but in person, so we didn't follow up here. That should work without problems for now.
Adding it to minifi.sh could be a good solution if there is a framework for it like you mentioned, but I am not sure everyone uses minifi.sh at the moment (I usually just start the binary).

Reply:
Instructions guide toward minifi.sh so that is the officially supported solution. I'll take a look at the factory initializer a little later. If there is a reason it can't be done in the shell scripts then I'm +1 to that, otherwise I'm a -1 to it.

Reply:
@phrocker I see your point, but I am a little afraid of adding functionality to the minifi.sh. When debugging, for example, you do not start to debug the shell running the minifi, but start debugging the minifi binary itself. If you do that, and the minifi.sh does not set the environment variables, it will have a different behaviour. I know that this is avoidable, but I am sure that more than one people would make this mistake.
Setting (and getting) environment variables from inside the process is perfectly supported, but not thread safe. Having the setenv in an ObjectFactoryInitializer, when the application is single threaded and the ObjectFactoryInitializers are called in sequence should not be a problem.
I do not think either solution is bad, but I feel that the ObjectFactoryInitializer would be better.

Reply:
@bakaid brought up a good point. We should update https://github.com/apache/nifi-minifi-cpp/blob/master/PROCESSORS.md as there are exceptional conditions, @nghiaxlee . Since you didn't create the processor will you need help generating this? Happy to help.

Reply:
@phrocker : Thank you, that would be great.